docs/rules/prefer-number-is-integer.md

```diff
@@ -0,0 +1,44 @@
+# Prefer `Number.isInteger()` for integer checking
+
```

Comment:
There are some new requirements for rule docs, be sure to rebase on the latest `main` to pick them up.